web/src/main/java/org/cbioportal/web/DataAccessTokenController.java

```diff
@@ -0,0 +1,123 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.web;
+
+import org.cbioportal.model.DataAccessToken;
+import org.cbioportal.service.DataAccessTokenService;
+import org.cbioportal.service.exception.DataAccessTokenNoUserIdentityException;
+import org.cbioportal.service.exception.DataAccessTokenProhibitedUserException;
+import org.cbioportal.web.config.annotation.InternalApi;
+
+import io.swagger.annotations.*;
+import java.util.*;
+import org.springframework.beans.factory.annotation.*;
+import org.springframework.http.*;
+import org.springframework.security.core.Authentication;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.HttpClientErrorException;
+
+@InternalApi
+@RestController
+@Validated
+@Api(tags = "Data Access Tokens", description = " ")
+public class DataAccessTokenController {
+
+    @Autowired
+    // use @Qualifier to ensure we get tokenService bean from applicationContext-security.xml
+    // tokenSerice bean in security.xml file has same name so would be picked anyway by default,
+    // but this avoids a NoUniqueBeanDefinitionException
+    @Qualifier("tokenService")
+    private DataAccessTokenService dataAccessTokenService;
+
+//      TODO: figure out how to read this from the properties file
+//    @Value("${security.data_tokens.unauth_users}")
+    private String[] USERS_WHO_CANNOT_USE_TOKENS = {"anonymousUser", "servcbioportal"};
+    private Set<String> usersWhoCannotUseTokenSet = null;
+
+    @Autowired
+    private void initializeUsersWhoCannotUseTokenSet() {
+        usersWhoCannotUseTokenSet = new HashSet<>(Arrays.asList(USERS_WHO_CANNOT_USE_TOKENS));
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/data-access-tokens", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<DataAccessToken> createDataAccessToken(Authentication authentication,
+                                    @RequestParam(required = false) Boolean allowRevocationOfOtherTokens) throws HttpClientErrorException {
+        DataAccessToken createdToken;
+        if (allowRevocationOfOtherTokens != null) {
+            createdToken = dataAccessTokenService.createDataAccessToken(getAuthenticatedUser(authentication), allowRevocationOfOtherTokens);
+        }
+        else {
+            createdToken = dataAccessTokenService.createDataAccessToken(getAuthenticatedUser(authentication));
+        }
+        if (createdToken == null) {
+            return new ResponseEntity<>(new DataAccessToken(null), HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(createdToken, HttpStatus.CREATED);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/data-access-tokens/fetch")
```

Comment:
You don't actually need `/fetch` for `GET` requests, it's for POST requests that sends a query object to get a result.